src/plugins/intel_cpu/src/emitters/snippets/cpu_runtime_configurator.cpp

```diff
@@ -69,6 +67,9 @@ void CPURuntimeConfigurator::update(const ov::snippets::lowered::LinearIRCPtr& l
     if (linear_ir->is_dynamic()) {
         update_loop_args(linear_ir);
     }
+    update_data_offsets();
+    m_final_runtime_optimizers.run(*linear_ir);
+    m_config->m_latest_shapes = std::move(m_config->shapes);
```

Comment:
I think we can keep `master_shape` `io_shapes` `io_layouts` update inside `update()`, since it aligns with `initialization()` for example.
My concern here is that `update` returns config with invalid state, i.e. stolen io_shapes. So I propose to move `latest_shapes` initialization (and io_shapes corruption) to the latest possible moment. In this case, it's just before the return from `get_updated_config`.